### PR TITLE
Add Article 6 & 9 Privacy Notice to Data Map Export

### LINKED ISSUE #368: Article 6 & 9 Privacy Notices
### Is your feature request related to a specific problem?

As part of exporting an Article 30 compliant RoPA, the articles 6 & 9 criteria must be documented and made available

### Describe the solution you'd like

In my simplistic view of the, world I would like to see enums for each attribute with a way to select them automatically based on selection from the default taxonomy. Not sure if this is possible

### Describe alternatives you've considered, if any

n/a

### Additional context

[Article 6](https://gdpr-info.eu/art-6-gdpr/) speaks to the lawful basis for processing personal data and is currently loosely modeled as part of the extended data use. However there may be room for improvement here so including a re-look at it as part of the privacy notice section

[Article 9](https://gdpr-info.eu/art-9-gdpr/) is the condition for processing special category data

### PR DESCRIPTION
**Pull request by SteveDMurphy**
Closes #368 

### Code Changes

* [x] Add a new column to `data_uses` for special categories
* [x] Add a new enum to model a `special_category` as part of a `DataUse`
* [x] Update the output of a system to contain a special category where defined
* [x] Update the exported data map to include both of the article 6 & 9 legal basis and special categories

### Steps to Confirm

* [x] export the demo_resources systems
* [x] export the demo_resources data map

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Documentation Updated

### Description Of Changes

Part of this change is relatively straight forward in adding the models, the area of opinion here may be the verbiage for both the legal basis and the special category that is included in the enum. Previously some mention of a user-defined enum has been discussed and these are the two most likely scenarios where this could be potentially applied.

The demo_resources have been updated to easily validate the changes.

To start, ensure you have the latest database migration applied and run the following:
```
$ fidesctl apply demo_resources
$ fidesctl export system demo_resources --dry
```

To see the full data map export (apply not `required` if run already with the above):
```
$ fidesctl apply demo_resources
$ fidesctl export datamap demo_resources
```

### CODE REVIEW

File: fidesctl/src/fidesctl/core/export.py [hunk 1]
```diff
@@ -126,13 +126,13 @@ def generate_system_records(
     server_system_list: List,
     url: str,
     headers: Dict[str, str],
-) -> List[Tuple[str, str, str, str, str, str, str, str, str, str, str, str]]:
+) -> List[Tuple[str, ...]]:
```
Comment (SteveDMurphy) [inline fidesctl/src/fidesctl/core/export.py:129]: I'm not sure this is an improvement, but it makes it a bit cleaner. Open to reverting to being very explicit about the length of the tuple if that is preferred 👍🏽

**Review by ThomasLaPiana — APPROVED** [review]
LGTM! Small model change, only thing to watch out for is alembic migration issues potentially

File: fidesctl/src/fidesctl/core/export.py (hunk 1 above)
Comment (SteveDMurphy) [reply fidesctl/src/fidesctl/core/export.py:129]: Thanks! That all makes sense to me 🙌🏽

**Comment by SteveDMurphy** [comment]
Holding off merging until GitHub has resolved the open status alert for GitHub Actions -> https://www.githubstatus.com/incidents/83lq7ftk19r5

**Pull request merged**